OpenRA.Mods.Common/Widgets/ButtonWidget.cs

```diff
@@ -251,13 +251,9 @@ public override void Draw()
 			var hover = Ui.MouseOverWidget == this || Children.Any(c => c == Ui.MouseOverWidget);
 			DrawBackground(rb, disabled, Depressed, hover, highlighted);
 
-			// Scissor when the text overflows
-			var isTextOverflowing = font.Measure(text).X > contentWidth;
-			if (isTextOverflowing && TooltipContainer != null)
+			// Truncate when the text overflows
+			if (font.Measure(text).X > contentWidth && TooltipContainer != null)
```

Comment:
This check is still dodgy in general: `text` may change between frames and if it hits this limit even once it will bind the tooltip for all future time.

Reply:
It would probably be better to revert the `ButtonWidget` changes completely and implement the truncation/tooltip logic in the `SettingsLogic` instead. You can guarantee the way your labels behave there, which saves you from having to guard against these more difficult dynamic cases.